src/test/java/microsoft/exchange/webservices/data/EwsXmlReaderTest.java

```diff
@@ -0,0 +1,77 @@
+package microsoft.exchange.webservices.data;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.events.Characters;
+import javax.xml.stream.events.XMLEvent;
+import java.io.ByteArrayInputStream;
+
+import static org.mockito.Mockito.doReturn;
+
+public class EwsXmlReaderTest {
+
+  @Mock(name="presentEvent") XMLEvent presentEvent;
+  @Mock(name="xmlReader") XMLEventReader xmlReader;
+  @InjectMocks EwsXmlReader impl;
+  @Mock Characters character;
+
+  @Before
+  public void setUp() throws Exception {
+    impl = new EwsXmlReader(new ByteArrayInputStream(("<?xml version=\"1.0\" encoding=\"UTF-8\"?>"
+        + "<test></test>").getBytes("UTF-8")));
+    MockitoAnnotations.initMocks(this);
+
+  }
+
+  @Test
+  public void testReadValueWhenCharacterDataIsNull() throws Exception {
+
+    doReturn(false).when(presentEvent).isStartElement();
```

Comment:
I usually write unit tests like this one:
- as instance variables, I declare all mocks needed for the test. The instance variable called 'impl' is *always* an instance of the class under test. All these instances variables *must* be refreshed before each test method execution.
- I declare a setUp method annotated with '@Before' which is in charge of refreshing all instances variables before each test and that contains as well all generic mock configuration (ie: all mock configuration that can be shared with all test methods)
- finally, I declare test methods where I follow the convention //given //when //then (see http://en.wikipedia.org/wiki/Behavior_Driven_Development).

In the case of this specific test, I expect all my mock declarations to be useful and needed to write other test methods. That's why it makes sens to me to keep this mocking code inside the test. 
But I'm obviously open to other suggestion!